projects/jdnconvertible-calendar-date-adapter/src/lib/jdnconvertible-calendar-date-adapter.ts

```diff
@@ -250,6 +250,15 @@ export class JDNConvertibleCalendarDateAdapter extends DateAdapter<JDNConvertibl
    * @returns a date in the specified calendar.
    */
   private createCalendarDate(year: number, month: number, date: number, calendar: string): JDNConvertibleCalendar {
+
+    if (month < 0 || month > 11) {
+      throw Error(`Invalid month index "${month}". Month index has to be between 0 and 11.`);
+    }
+
+    if (date < 1) {
+      throw Error(`Invalid date "${date}". Date has to be greater than 0.`);
```

Comment:
the error message does not make sense with the announced condition

Reply:
I actually copied this from https://github.com/angular/components/blob/431678730b1070b194e9b51ad29bb8a506881c9e/src/material-moment-adapter/adapter/moment-date-adapter.ts#L157-L178 :-)

What do you mean by "announced condition"?

Reply:
if date <1 so the date is greater than 0. I do not understand the 2nd sentence here.

Reply:
`date` is actually the day of the month (that confused me). The check just makes sure that the day is at least 1, it throws an error if you pass 0 or a negative number

Reply:
so why the message is not "Date has to be greater than 1."?

Reply:
sorry, I am not following. The error will be thrown if 0 is passed or any negative number. It won't be thrown for 1, 2 etc.

Reply:
so the date must be at least 1 and it is an integer

Reply:
yes, but there is no check whether the number has fractions (in fact, TS/JS only know the number type)

Reply:
I think TS offers `bigint` but I've never used it

Reply:
ok, I improved the error message in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/27/commits/2ea9ac008776b5c84eef62be687b1f8c140cfbd3

Reply:
some tests added in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/27/commits/d150b8f26bd3ee7d896bf7805e3d5c07ec822d51